compiler/p/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1693,864 +2024,786 @@ TR::Register *OMR::Power::TreeEvaluator::ifacmpneEvaluator(TR::Node *node, TR::C
    return NULL;
    }
 
+void generateCompareBranchSequence(
+   TR::Register *trgReg,
+   TR::Node *node,
+   TR::Node *firstChild,
+   TR::Node *secondChild,
+   const CompareInfo& compareInfo,
+   TR::CodeGenerator *cg
+)
+   {
+   TR::Register *condReg = cg->allocateRegister(TR_CCR);
+   CompareCondition cond = evaluateCompareToConditionRegister(condReg, node, firstChild, secondChild, compareInfo, cg);
+
+   TR::LabelSymbol *startLabel = generateLabelSymbol(cg);
+   startLabel->setStartInternalControlFlow();
 
-// also handles acmpeq in 32-bit mode
-// and also: bcmpeq, scmpeq
-TR::Register *OMR::Power::TreeEvaluator::icmpeqEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   TR::LabelSymbol *endLabel = generateLabelSymbol(cg);
+   endLabel->setEndInternalControlFlow();
+
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::li, node, trgReg, 0);
+   generateLabelInstruction(cg, TR::InstOpCode::label, node, startLabel);
+   generateConditionalBranchInstruction(cg, compareConditionToBranch(reverseCondition(cond)), node, endLabel, condReg);
+   generateTrg1ImmInstruction(cg, TR::InstOpCode::li, node, trgReg, 1);
+
+   TR::RegisterDependencyConditions *deps = new (cg->trHeapMemory()) TR::RegisterDependencyConditions(0, 2, cg->trMemory());
+   deps->addPostCondition(condReg, TR::RealRegister::NoReg);
+   deps->addPostCondition(trgReg, TR::RealRegister::NoReg);
+
+   generateDepLabelInstruction(cg, TR::InstOpCode::label, node, endLabel, deps);
+
+   cg->stopUsingRegister(condReg);
+   }
+
+void generateCompareSetBoolean(
+   TR::Register *trgReg,
+   TR::Node *node,
+   TR::Node *firstChild,
+   TR::Node *secondChild,
+   const CompareInfo& compareInfo,
+   TR::CodeGenerator *cg
+)
    {
-   TR::Register *trgReg    = cg->allocateRegister();
-   TR::Node     *firstChild   = node->getFirstChild();
-   TR::Node     *secondChild  = node->getSecondChild();
-   TR::Register *src1Reg = cg->evaluate(firstChild);
-   TR::Register *temp1Reg     = NULL;
-   TR::Register *temp2Reg     = cg->allocateRegister();
-   bool         killTemp1    = false;
+   TR::Register *condReg = cg->allocateRegister(TR_CCR);
+   CRCompareCondition cond = compareConditionInCR(evaluateCompareToConditionRegister(condReg, node, firstChild, secondChild, compareInfo, cg));
+
+   generateTrg1Src1ImmInstruction(cg, cond.isReversed ? TR::InstOpCode::setbcr : TR::InstOpCode::setbc, node, trgReg, condReg, cond.crcc);
 
```

Comment:
cond.crcc << pos_RA?  crcc alone is wrong.

Reply:
The `setbc` family of opcodes do not use the same hack that the CR logical instructions do, so the proper shift is applied during binary encoding not here. I had plans to refactor the way that the fields are specified for CR logical instructions to make it less awkward to use, but that has not come to fruition yet.